core/src/main/kotlin/org/evomaster/core/problem/graphql/service/GraphQLFitness.kt

```diff
@@ -257,24 +258,114 @@ class GraphQLFitness : HttpWsFitness<GraphQLIndividual>() {
-       val query = "{${selection.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}"
+        if (a.methodType == GQMethodType.QUERY) {
+
+            if (inputGenes.isNotEmpty()) {
+
+                val printableInputGene: MutableList<String> = getPrintableInputGene(inputGenes)
+
+                var printableInputGenes = getPrintableInputGenes(printableInputGene)
+
+
+                if (returnGene.name.toLowerCase() == "scalar") {
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}  ($printableInputGenes)         } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+
+                } else {
+
+                    var query = getQuery(returnGene, a)
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}  ($printableInputGenes)  $query       } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                }
+            } else {
+                if (returnGene.name.toLowerCase() == "scalar") {
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}       } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                } else {
+                    var query = getQuery(returnGene, a)
+                    val bodyEntity = Entity.json("""
+            {"query" : "  { ${a.methodName}  $query     } ","variables":null}
+        """.trimIndent())
+                    val invocation = builder.buildPost(bodyEntity)
+                    return invocation
+                }
+            }
+        } else if (a.methodType == GQMethodType.MUTATION) {
+            val printableInputGene: MutableList<String> = getPrintableInputGene(inputGenes)
+
+            var printableInputGenes = getPrintableInputGenes(printableInputGene)
 
-        val bodyEntity = Entity.json("""
-            {"query" : "$query","variables":null}
+            if (returnGene.name.toLowerCase() == "scalar") {
+                val bodyEntity = Entity.json("""
+            {"query" : " mutation{ ${a.methodName}  ($printableInputGenes)         } ","variables":null}
         """.trimIndent())
+                val invocation = builder.buildPost(bodyEntity)
+                return invocation
 
-        val invocation = builder.buildPost(bodyEntity)
+            } else {
+                var mutation = getMutation(returnGene, a)
+
+                val bodyEntity = Entity.json("""
+            { "query" : "mutation{    ${a.methodName}  ($printableInputGenes)    $mutation    }","variables":null}
+        """.trimIndent())
+                val invocation = builder.buildPost(bodyEntity)
+                return invocation
```

Comment:
this
````
val invocation = builder.buildPost(bodyEntity)
return invocation
```
is repeated a lot of times. could be simplified (eg taken out after the if statements)

Reply:
done